pages/login.tsx

```diff
@@ -1,55 +1,147 @@
 
-import Layout from '@components/Layout'; // Layout wrapper
 import useLocalStorage from '@tools/hooks/useLocalStorage'; // Local storage
 
 export default function Login() {
   const [email, setEmail] = useState(''); // Email input
+  const [emailInputError, setEmailInputError] = useState(''); // Error message displayed under input
 
   // Store email in local storage
   const [, setLocalStorageEmail] = useLocalStorage('rcd-email-redirect', '');
 
   // Loading state for log in button
   const [isSigningIn, setIsSigningIn] = useState(false);
+  // NextAuth signIn() resolved promise
+  const [authState, setAuthState] = useState<SignInResponse>();
 
   /**
    * Process login using email input
    */
-  const signInWithEmail = () => {
+  const signInWithEmail = async () => {
     setIsSigningIn(true);
     setLocalStorageEmail(email);
-    signIn('email', { email });
+    const signInResponse = await signIn('email', { email, redirect: false });
+    // Store NextAuth promise response
+    setAuthState(signInResponse);
+    setIsSigningIn(false);
+
+    if (signInResponse?.error) {
+      setEmailInputError('This email has not been registered by the admin.');
+    }
   };
 
   const handleSubmit = (event: SyntheticEvent) => {
     event.preventDefault();
-    signInWithEmail();
+    if (email.length) {
+      signInWithEmail();
+    } else {
+      setEmailInputError('Please enter an email address.');
+    }
   };
 
+  const resendEmailToast = useToast();
+
   return (
-    <Layout>
-      <Container>
-        <form onSubmit={handleSubmit}>
-          <FormControl>
-            <FormLabel>Login</FormLabel>
-            <Input
-              type="email"
-              placeholder="Your email"
-              isDisabled={isSigningIn}
-              value={email}
-              onChange={event => setEmail(event.target.value)}
-            />
-            <FormHelperText>Please enter an RCD email</FormHelperText>
-          </FormControl>
-          <Button onClick={signInWithEmail} isLoading={isSigningIn} loadingText="Logging In">
-            Log in
-          </Button>
-        </form>
-      </Container>
-    </Layout>
+    <Center minHeight="100vh" bg="background.grey">
+      <Box
+        borderWidth="1px"
+        borderRadius="12px"
+        bg="background.white"
+        height="60%"
+        width="37.5%"
+        paddingBottom="3%"
+        overflow="hidden"
+      >
+        <Container maxWidth="82.5%">
+          <VStack width="100%" spacing={12} marginTop="5%">
+            <Flex direction="column" style={{ gap: '10px' }}>
+              <Image src="/assets/rcd_logo.svg" height={120} width={120} />
+              <Text as="h1" textStyle="display-medium" align="center">
+                Richmond Centre for Disability Employee Login
+              </Text>
+            </Flex>
+
+            {/* If NextAuth's callback doesn't contain a URL, auth (sending email) wasn't successful. */}
+            {!authState?.url ? (
```

Comment:
I was reading up on the [nextAuth docs](https://next-auth.js.org/getting-started/client#:~:text=signin%20will%20then%20return%20a%20promise%2C%20that%20resolves%20to%20the%20following%3A) and was just curious - why did you use `authState.url` and not `authState.ok` here?

Reply:
`authState.ok` will always contain `200` (the HTTP code) if the user's login request is succesfully received by the NextAuth client, regardless of whether or not the user has a valid email. We need to check for `url` instead since it only gets returned when an actual login email gets sent out.